### PR TITLE
sanitycheck: fail on warnings

### PR DESCRIPTION
**Pull request by nashif**

### CODE REVIEW

File: scripts/sanitycheck [hunk 1]
```diff
@@ -1777,8 +1777,16 @@ class CMake():
...
+        ldflags = ""
+        cflags = ""
+
+        ldflags="-Wl,--fatal-warnings"
+
         cmake_args = ['-B{}'.format(self.build_dir),
                       '-S{}'.format(self.source_dir),
+                      '-DEXTRA_CFLAGS="-Werror {}'.format(cflags),
+                      '-DEXTRA_AFLAGS=-Wa,--fatal-warnings',
+		      '-DEXTRA_LDFLAGS="{}'.format(ldflags),
```
Comment (galak) [inline scripts/sanitycheck]: Nit, feels like white space isn't aligned.

File: scripts/sanitycheck [hunk 2]
```diff
@@ -1777,8 +1777,16 @@ class CMake():
 
         verbose("Running cmake on %s for %s" %(self.source_dir, self.platform.name))
 
+        ldflags = ""
```
Comment (galak) [inline scripts/sanitycheck]: seems like have aflags here as well.

File: scripts/sanitycheck [hunk 3]
```diff
@@ -1777,9 +1777,17 @@ class CMake():
 
         verbose("Running cmake on %s for %s" %(self.source_dir, self.platform.name))
 
-        cmake_args = ['-B{}'.format(self.build_dir),
-                      '-S{}'.format(self.source_dir),
-                      '-G{}'.format(get_generator()[1])]
+        ldflags="-Wl,--fatal-warnings"
```
Comment (galak) [inline scripts/sanitycheck]: why is ldflags special?

**Review by aescolar — APPROVED** [review]

**Comment by aescolar** [comment]
Fixes #20025

**Comment by aescolar** [comment]
Fixes #20049

File: scripts/sanitycheck (hunk 3 above)
Comment (nashif) [reply scripts/sanitycheck]: it is not, I copied the code as it was in previous version and modified it slightly while I look into a long term fix and addressing some other options we have to deal with errors and deprecations.

**Pull request merged**